### ISSUE TITLE
Error serialize FileField None.

### ISSUE THREAD

**Issue by koyoyo** [pennersr/django-allauth#1577]
When connect with social (eg. Facebook) for new user, it'll error when try to serialize image if has None value. it'll have stack trace like this

```
allauth/socialaccount/helpers.py in complete_social_login
allauth/socialaccount/helpers.py in _complete_social_login
allauth/socialaccount/helpers.py in _process_signup
allauth/socialaccount/models.py in serialize
allauth/socialaccount/adapter.py in serialize_instance
allauth/utils.py in serialize_instance

def serialize_instance(instance):
    data = {}
    for k, v in instance.__dict__.items():
        if k.startswith('_') or callable(v):
            continue
        try:
            field = instance._meta.get_field(k)
            if isinstance(field, BinaryField):
                v = force_text(base64.b64encode(v))
            elif isinstance(field, FileField):
                if not isinstance(v, six.string_types):
                    v = v.name
            # Check if the field is serializable. If not, we'll fall back
            # to serializing the DB values which should cover most use cases.
            try:
                json.dumps(v, cls=DjangoJSONEncoder)
            except TypeError:
                v = field.get_prep_value(v)
                k = SERIALIZED_DB_FIELD_PREFIX + k
        except FieldDoesNotExist:
            pass
        data[k] = v
    return json.loads(json.dumps(data, cls=DjangoJSONEncoder))
```

In this function, it checks that if it's instance of FileField and that data is not string type `not isinstance(v, six.string_types)` for calling `v = v.name` to serialize file data. But if v is None, it's still do this (None is not string type), then error.

Note: `django-allauth==0.29.0`

**Comment by pennersr** [comment]
Already fixed in 68b73ff91e15a484186ca8e81e3b8d9d55828bab

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/1577
